user-service/src/main/java/com/movie/recommender/user/service/UserService.java

```diff
@@ -55,4 +59,43 @@ public String login(String username, String password) {
             throw new IllegalArgumentException("Invalid credentials");
         }
     }
+
+    public Set<Long> getFavoriteMovies(Long userId) {
+        log.info("Attempting to get favorite movies for user with id: {}", userId);
+        User user = userRepository.findById(userId)
```

Comment:
extract exception throwing logic into a helper method to avoid repetition (something like `findUserById`)